packages/components/button/src/Button.tsx

```diff
@@ -10,14 +12,32 @@ export interface ButtonProps
    * Change the component to the HTML tag or custom component of the only child.
    */
   asChild?: boolean
+  /**
+   * Display a spinner to indicate to the user that the button is loading something after they interacted with it.
+   */
+  isLoading?: boolean
+  /**
+   * If your loading state should only display a spinner, it's better to specify a label for it (a11y).
+   */
+  loadingLabel?: string
+  /**
+   * If your loading state should also display a label, you can use this prop instead of `loadingLabel`.
+   * **Please note that using this can result in layout shifting when the Button goes from loading state to normal state.**
+   */
+  loadingText?: string
```

Comment:
Does it make sense to pass both props (loadingLabel and loadingText) simultaneously?

If not, shouldn't we ensure (using TS) that it's not possible to do so?

Reply:
You should pass one or the other.

It's a nightmare to do this kind of thing with TS. It means a huge union type of two interfaces.

Reply:
It's not that bad actually 😛,
```tsx
type CustomProps =  | {
      asChild?: boolean
      isLoading?: boolean
      loadingText?: string
      loadingLabel?: never
    }
  | {
      asChild?: boolean
      isLoading?: boolean
      loadingLabel?: string
      loadingText?: never
    }

type ButtonProps = PropsWithChildren<
  Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'disabled'>
> &
  ButtonStylesProps &
  Propse
```

Reply:
But I understand if you think it's not worth adding this TS complexity, as it doesn't add that much value in the end 👍